libraries/AP_Winch/AP_Winch_Servo.cpp

```diff
@@ -0,0 +1,92 @@
+#include <AP_Winch/AP_Winch_Servo.h>
+
+extern const AP_HAL::HAL& hal;
+
+void AP_Winch_Servo::init(const AP_WheelEncoder* wheel_encoder)
+{
+    _wheel_encoder = wheel_encoder;
+
+    // set servo output range
+    SRV_Channels::set_angle(SRV_Channel::k_winch,  1000);
+}
+
+void AP_Winch_Servo::update()
+{
+    // return immediately if no servo is assigned to control the winch
+    if (!SRV_Channels::function_assigned(SRV_Channel::k_winch)) {
+        return;
+    }
+
+    // return immediately if no wheel encoder
+    if (_wheel_encoder == nullptr) {
+        return;
+    }
+
+    // if not doing any control output trim value
+    if (config.state == AP_Winch::STATE_RELAXED) {
+        SRV_Channels::set_output_limit(SRV_Channel::k_winch, SRV_Channel::SRV_CHANNEL_LIMIT_TRIM);
+        return;
+    }
+
+    // calculate dt since last iteration
+    uint32_t now = AP_HAL::millis();
+    float dt = (now - last_update_ms) / 1000.0f;
+    if (dt > 1.0f) {
+        dt = 0.0f;
+    }
+    last_update_ms = now;
+
+    // calculate latest rate
+    float distance = _wheel_encoder->get_distance(0);
+    float rate = 0.0f;
+    if (is_positive(dt)) {
+        rate = (distance - _last_distance) / dt;
+    }
+    _last_distance = distance;
+
+    // update distance from wheel encoder
+    config.length_curr = distance;
```

Comment:
`_last_distance` and `config.length_curr` always track the same value, can `_last_distance` be dropped in favour of the other?

Reply:
nice optimisation.  dropping _last_distance.